File: cuckoo-hibersap-example-datamodel/src/main/java/de/akquinet/jbosscc/cuckoo/example/model/Severity.java
package de.akquinet.jbosscc.cuckoo.example.model;

public enum Severity
{
    INFO( 'I' ), SUCCESS( 'S' ), WARNING( 'W' ), ERROR( 'E' ), ABORT( 'A' ), OTHER( ' ' );

    private char sapType;

    Severity( char sapType )
    {
        this.sapType = sapType;
    }

    public static Severity fromSapType( char sapType )
    {
        for ( Severity severity : Severity.values() )
        {
            if ( severity.sapType == sapType )
            {
                return severity;
            }
        }
        return OTHER;
    }
}

File: src/main/kotlin/org/hibersap/customer/model/Severity.kt
package org.hibersap.customer.model

enum class Severity(val sapType: Char) {
    INFO('I'), SUCCESS('S'), WARNING('W'), ERROR('E'), ABORT('A'), OTHER(' ');

    companion object {
        fun fromSapType(sapType: Char): Severity {
            for (severity in values()) {
                if (severity.sapType == sapType) {
                    return severity
                }
            }
            return OTHER
        }
    }
}